lib/backpex/resource.ex

```diff
@@ -40,35 +40,37 @@ defmodule Backpex.Resource do
   @doc """
   Gets a database record with the given `fields` by the given  `primary_value`.
 
-  Raises `Ecto.NoResultsError` if no record was found.
+  Returns `{:ok, nil}` if no result was found.
 
   ## Parameters
 
   * `primary_value`: The identifier for the specific item to be fetched.
   * `assigns` (map): The current assigns of the socket.
   * `live_resource` (module): The `Backpex.LiveResource` module.
   """
-  def get!(primary_value, assigns, live_resource) do
+  def get(primary_value, assigns, live_resource) do
     adapter = live_resource.config(:adapter)
 
-    adapter.get!(primary_value, assigns, live_resource)
+    adapter.get(primary_value, assigns, live_resource)
   end
 
   @doc """
   Gets a database record with the given `fields` by the given  `primary_value`.
 
-  Returns `nil` if no result was found.
+  Raises `Backpex.NoResultsError` if no record was found.
 
   ## Parameters
 
   * `primary_value`: The identifier for the specific item to be fetched.
   * `assigns` (map): The current assigns of the socket.
   * `live_resource` (module): The `Backpex.LiveResource` module.
   """
-  def get(primary_value, assigns, live_resource) do
-    adapter = live_resource.config(:adapter)
-
-    adapter.get!(primary_value, assigns, live_resource)
+  def get!(primary_value, assigns, live_resource) do
+    case get(primary_value, assigns, live_resource) do
+      {:ok, nil} -> raise Backpex.NoResultsError
+      {:ok, result} -> result
+      {:error, _error} -> raise Backpex.NoResultsError
+    end
```

Comment:
@Flo0807 Like this? I think we should work with result tuples everywhere (like Ash for example).

Reply:
👍